### PR TITLE
Recover from a corrupt config on startup

### PR DESCRIPTION
**Pull request by xoriole**
If Tribler encounters a corrupted config file during startup, it should try to recover using the default config and inform the user about it. This is exactly what this PR does.

Fixes https://github.com/Tribler/tribler/issues/5548

Screenshot
Screenshot 2021-01-08 at 12 45 53

### CODE REVIEW

**Review by ichorid — COMMENTED** [review]
The test config file should be full of binary zeroes, not just text `\00` :wink:

**Comment by drew2a** [comment]
@xoriole probably you have to change the target branch to release-7.7 :)

**Review by ichorid — CHANGES_REQUESTED** [review]
The test config file is still not binary zeroes. That's ASCII zeroes.
изображение

File: src/run_tribler.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ async def start_tribler():
 
         state_dir = get_versioned_state_directory(root_state_dir)
 
-        config = TriblerConfig(state_dir, config_file=state_dir / CONFIG_FILENAME)
+        config = TriblerConfig(state_dir, config_file=state_dir / CONFIG_FILENAME, recover_error=True)
```
[the remark this answers is not in the document]
Comment (xoriole) [reply src/run_tribler.py]: Done

File: src/tribler-core/tribler_core/config/test_tribler_config.py [hunk 2]
```diff
@@ -13,6 +21,21 @@ def test_init_without_config(tribler_config):
...
+    # By default, recover_error set to False when loading the config file so
+    # if the config file is corrupted, it should raise a ParseError.
+    with pytest.raises(ParseError):
+        _ = TriblerConfig(tmpdir, config_file=corrupt_config_file)
+
+    # If recover_error is set to True, the config should successfully load using
+    # the default config in case of corrupted config file and the error is saved.
+    tribler_conf = TriblerConfig(tmpdir, config_file=corrupt_config_file, recover_error=True)
+    assert "configobj.ParseError: Invalid line" in tribler_conf.config_error
+
```
[the remark this answers is not in the document]
Comment (xoriole) [reply src/tribler-core/tribler_core/config/test_tribler_config.py:38]: Done

**Comment by ichorid** [comment]
:+1:

**Review by ichorid — APPROVED** [review]

**Pull request merged**